scripts/wrist_test.py

```diff
@@ -0,0 +1,298 @@
+
+            el_axis1_targets = np.linspace(ranges[-1], ranges[0], samples)
+
+            for t2 in el_axis1_targets:
+                msg.name = ["elbow_" + left_or_right + "_axis0"]
+                msg.position = [t2]
+                msg.velocity = [0]
+                msg.effort = [0]
+                publisher.publish(msg)
+                rate.sleep()
+
+
+def move_shoulder_elbow_wrist(left_or_right, move_elbow=True, move_wrist=True,
+                              shoulder_range=None, elbow_range=None, wrist_range=None, samples=200):
+
+    msg = JointState()
+
+    msg.name = ["shoulder_" + left_or_right + "_axis0"]
+    targets = [np.linspace(shoulder_range[0], shoulder_range[1], samples)]
+    if move_elbow:
+        msg.name.append("elbow_" + left_or_right + "_axis0")
+        targets.append(np.linspace(elbow_range[0], elbow_range[1], samples))
+    if move_wrist:
+        msg.name.append("wrist_" + left_or_right + "_axis0")
+        targets.append(np.linspace(wrist_range[0], wrist_range[1], samples))
+
+    targets = np.array(targets)
+
+    for i in range(samples):
+        msg.position = list(targets[:, i])
+        msg.velocity = [0] * len(msg.name)
+        msg.effort = [0] * len(msg.name)
+        publisher.publish(msg)
+        rate.sleep()
+
+
+def move_wrist(left_or_right, axis, ranges, samples=200):
+    msg = JointState()
+    msg.name = ["wrist_" + left_or_right + "_axis" + str(axis), "wrist_" + left_or_right + "_axis0"]
+    msg.velocity = [0] * len(msg.name)
+    msg.effort = [0] * len(msg.name)
+
+    for i in range(6):
+
+        if not i % 2:
+            targets = np.linspace(ranges[-1], ranges[0], samples)
+
+            for t in targets:
+                msg.position = [t, t]
+                publisher.publish(msg)
+                rospy.sleep(0.01)
+
+        else:
+            targets = np.linspace(ranges[0], ranges[-1], samples)
+
+            for t in targets:
+                msg.position = [t, t]
+                publisher.publish(msg)
+                rospy.sleep(0.01)
+
+
+def move_all(move_style=MoveStyle.SEQUENTIALLY,
+             shoulder_range=None, shoulder_samples=200,
+             elbow_range=None, elbow_samples=200,
+             wrist_range=None, wrist_samples=200,
+             head_range=None, head_samples=200):
+    msg = JointState()
+    msg.name = ["shoulder_left_axis0", "elbow_left_axis0", "wrist_left_axis0", "wrist_left_axis1",
+                "shoulder_right_axis0", "elbow_right_axis0", "wrist_right_axis0", "wrist_right_axis1",
+                "head_axis0"]
+
+    if move_style == MoveStyle.SEQUENTIALLY:
+        targets = np.linspace(shoulder_range[0], shoulder_range[1], shoulder_samples)
+        targets_el = np.linspace(elbow_range[0], elbow_range[1], elbow_samples)
+        targets_wr = np.linspace(wrist_range[0], wrist_range[1], wrist_samples)
+        targets_head = np.linspace(head_range[0], head_range[1], head_samples)
+    elif move_style == MoveStyle.RANDOMLY:
```

Comment:
check if simulated == true here and log a warning or error and ignore the joint targets

Reply:
I already put the checking for simulation in the test_scene_4 (line 246). In my opinion, the move_all function acts like an abstract that cover all move style, so I decided to let the caller (test_scene_4) make the decision (via is_simulated param) instead.

Reply:
yes, i saw it, but you should have a check here, cause there might be other ppl using your script and your code must be idiot-proof, so right now technically I can still call this function with random movements with a physical robot.

Reply:
Updated